File: LeetCode/Medium/Reverse Words In A String/Go/reverse_words_in_a_string.go
package main

import (
	"fmt"
	"strings"
)

func reverse_words_in_a_string(s string) string {
	str := strings.Split(s, " ")
	i, k := 0, 0
	for i < len(str) {
		if str[i] != "" {
			str[k] = str[i]
			k++
		}
		i++
	}
	str = str[:k]
	for i := 0; i < k/2; i++ {
		str[i], str[k-1-i] = str[k-1-i], str[i]
	}
	return strings.Join(str, " ")
}

func main() {
	fmt.Println(reverse_words_in_a_string("the sky is blue"))
}

File: LeetCode/Medium/Reverse Words In A String/Cpp/reverse_words_in_a_string.cc
#include <iostream>
#include <vector>
#include <sstream>

std::string reverse_words_in_a_string(std::string s) {
    std::stringstream ss(s);
    std::string word;
	std::string res = "";
    while(ss>>word) {
        res = word + " " + res;
    }
    return res.substr(0, res.size()-1);
}

int main() {
    std::cout<<reverse_words_in_a_string("the sky is blue")<<"\n";
}
